lib/posix/fs.c

```diff
@@ -54,17 +57,38 @@ static inline void posix_fs_free_obj(struct posix_fs_desc *ptr)
 	ptr->used = false;
 }
 
+static int posix_mode_to_zephyr(int mf)
+{
+	int mode = 0;
+
+	mode = (mf & O_ACCMODE) + 1;
+
+	if (mf & O_CREAT) {
+		if (!(mode & FO_WRITE)) {
+			errno = EINVAL;
+		} else {
+			mode |= FO_CREATE;
+		}
+	}
+	return mode;
+}
+
```

Comment:
What about `O_APPEND` (which is defined in newlib, but missing from zephyr's minimal libc).  This function makes numeric assumptions about the values of O_RDWR etc; it would be more robust to specifically check flags, e.g.:
```
if (mf & O_RDWR)
    mode = FO_RDWR;
else if ...
```

Reply:
Done